test/kubemark/start-kubemark.sh

```diff
@@ -275,32 +345,63 @@ if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
     export TENANT_PARTITION_SEQUENCE=${tp_num}
     create-kubemark-master
 
-    if [[ "${KUBERNETES_SCALEOUT_PROXY}" == "true" ]]; then
-      export KUBERNETES_SCALEOUT_PROXY=false
-      PROXY_RESERVED_IP=$(grep server "${PROXY_KUBECONFIG}" | awk -F "/" '{print $3}' | cut -d: -f1)
-      export PROXY_RESERVED_IP
+    export PROXY_RESERVED_IP=$(cat ${KUBE_TEMP}/proxy-reserved-ip.txt)
+    echo "DBG: PROXY_RESERVED_IP=$PROXY_RESERVED_IP"
+    
+    export TP_${tp_num}_RESERVED_IP=$(cat ${KUBE_TEMP}/master_reserved_ip.txt)
+
+    # TODO: fix the hardcoded path
+    # the path is what the controller used in master init script on the master machines
+    if [[ ${tp_num} == 1 ]]; then
+      export TENANT_SERVERS="/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
+    else
+      export TENANT_SERVERS="${TENANT_SERVERS},/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
     fi
 
-    TP_SERVER="http://$(cat /tmp/master_reserved_ip.txt):8080"
+    ## TODO: cleanup the tmp config and reuse the local-kubeconfig
+    #
+    cp -f ${LOCAL_KUBECONFIG} "${RESOURCE_DIRECTORY}/kubeconfig.kubemark.tp-${tp_num}"
+
     if [[ ${tp_num} == 1 ]]; then
-      export TENANT_SERVERS=${TP_SERVER}
+      MASTER_METADATA="tp-${tp_num}=${RESOURCE_DIRECTORY}/kubeconfig.kubemark.tp-${tp_num}"
     else
-      export TENANT_SERVERS="${TENANT_SERVERS},${TP_SERVER}"
+      MASTER_METADATA=${MASTER_METADATA},"tp-${tp_num}=${RESOURCE_DIRECTORY}/kubeconfig.kubemark.tp-${tp_num}"
```

Comment:
I'm really not sure what's going on here.

Reply:
the MASTER_METADATA is used in the cloud-init i GCE where VM is created. in our case the master-init will start the k8s components during the master init so the info, in this case, the TP kubeconfig, is passed into the RP VM to start the kube-controller-manager process with the TP kubeconfig.